drivers/cache/cache_handlers.c

```diff
@@ -7,30 +7,86 @@
 #include <zephyr/cache.h>
 #include <zephyr/syscall_handler.h>
 
-static inline int z_vrfy_sys_cache_data_all(int op)
+static inline int z_vrfy_sys_cache_data_flush_all(void)
 {
-	return z_impl_sys_cache_data_all(op);
+	return z_impl_sys_cache_data_flush_all();
 }
-#include <syscalls/sys_cache_data_all_mrsh.c>
+#include <syscalls/sys_cache_data_flush_all_mrsh.c>
 
-static inline int z_vrfy_sys_cache_data_range(void *addr, size_t size, int op)
+static inline int z_vrfy_sys_cache_instr_flush_all(void)
+{
+	return z_impl_sys_cache_instr_flush_all();
+}
+#include <syscalls/sys_cache_instr_flush_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_data_invd_all(void)
+{
+	return z_impl_sys_cache_data_invd_all();
+}
+#include <syscalls/sys_cache_data_invd_all_mrsh.c>
+
+static inline int z_vrfy_sys_cache_instr_invd_all(void)
+{
+	return z_impl_sys_cache_instr_invd_all();
+}
```

Comment:
Wow, we allowing userspace to invalidate any part of cache - even if it contains kernel data / other userspace threads data.
We should newer allow this. Now we have perfectly valid syscall to break the kernel.

I'd propose to drop all of these syscalls - from my we shouldn't expose such functionality to userspace.

Reply:
> Now we have perfectly valid syscall to break the kernel.

We can drop the `*_all` operations but the `*_range` operations are going to be hopefully fine thanks to `Z_OOPS(Z_SYSCALL_MEMORY_WRITE(addr, size))`.

> I'd propose to drop all of these syscalls - from my we shouldn't expose such functionality to userspace.

This was discussed briefly in #50747 (see https://github.com/zephyrproject-rtos/zephyr/issues/50747#issuecomment-1263233911) and on discord, and the conclusion was that having these as syscalls can be useful is some circumstances.

So, what about dropping only the `*_all` syscalls?

Reply:
Or just drop `sys_cache_data_invd_all()`, which is the only dangerous one as I see it.  Flushes are invisible except for performance artifacts.

Or just embrace the complexity, do it the hard way, and implement _all() operations in syscalls by enumerating the memory partitions mapped to the thread.  That wouldn't be too terribly hard, I don't think.  But it's implementing a useful API, especially on architectures like Xtensa where lazy apps are going to have a hard time tracking all their cache polllution and want an easy button.

It's also worth pointing out that in the modern world of Spectre-evolved vulnerabilities, in the long term cache lines containing kernel data are likely going to be isolated from userspace at the hardware level anyway.

Reply:
> So, what about dropping only the *_all syscalls?

That would be good start, so let's definitely do that :)